app/views/play/CampaignView.coffee

```diff
@@ -1433,6 +1449,9 @@ module.exports = class CampaignView extends RootView
     if what is 'codeplay-ads'
       return !me.finishedAnyLevels() && serverConfig.showCodePlayAds && !features.noAds && me.get('role') isnt 'student'
 
+    if what is 'promotion'
+      return me.finishedAnyLevels() and not features.noAds and not isStudentOrTeacher and me.get('country') is 'united-states' and me.get('preferredLanguage', true) is 'en-US' and new Date() < new Date(2019, 11, 20)
```

Comment:
Wanted to confirm my understanding - It seems we will be showing the promotion modal the first time the user goes to the campaign view after completing any number of levels (due to `me.finishedAnyLevels()`), however since home users navigate to unit map after every level, the modal will show only after first level which is what we need - is that right?

Also, a type for the date check, it seems it should be december as per the PR description.

Reply:
Yes, after the first level.

JavaScript dates using this constructor have months zero-indexed and days one-indexed (loooool), so this is the correct date.